### ISSUE TITLE
Jumping in water breaks player water interaction

### ISSUE THREAD

**Issue by Xiernix** [FluffyFishGames/AdvancedCompany#52]
**Describe the bug**
Jumping in water that is high enough to slow you down but not fully submerge you or jumping straight into water deep enough to submerge you stops most water effects on the player. Water will no longer slow you, stop stamina regen, drain stamina while moving, or stop you from sprinting in it. It also stops the sound effects that play when walking through water. Sound effects for being fully submerged and drowning still work as intended.

**To Reproduce**

1. Start the game and route to March.
2. Go straight out the doors of the ship to the chasm filled with water located between the ship and main entrance.
3. Jump down into the water at the bottom of the chasm.
4. You should move at normal speed with no stamina drain, sprint, and regen stamina while under water.

Alternatively:

1. Go to a flooded moon.
2. Wait for the water to rise high enough to affect movement.
3. Stand in the water and jump.
4. The water will no longer slow you down, drain stamina, etc.

**Expected behavior**
Water should stop players from sprinting, stop stamina regen, cause stamina to drain while walking through it, and play the splashing sound effect while in it.

**Logfiles**
[Player.log](https://github.com/FluffyFishGames/AdvancedCompany/files/14092791/Player.log)

**Add a list of all of your mods**
BepInEx-BepInExPack-5.4.2100
Rune580-LethalCompany_InputUtils-0.6.1
PotatoePet-AdvancedCompany-1.0.143

**Comment by Xiernix** [comment]
After some extra testing I found that this bug also effects quicksand. It still drains stamina and pulls you in like normal but you'll be able to jump while in it as long as you have enough stamina allowing you to reset the time it takes to sink in and die. The bug will persist between moons until the player dies.

The flippers will also trigger the bug if you sprint while bobbing up and down on the waters surface with them but it won't really be noticeable unless they're unequipped. As long as the flippers are equipped they seem to work normally. Stamina will constantly drain and movement speed will be slower (still slightly faster than moving in water without them) which I assume are both intended.

Also while not a persistent I found that if you remove the flippers while under water it will also allow you to sprint and regen stamina underwater. The only difference being that it only lasts until you leave the water instead of until you die.

Log file from testing the above in case there's anything useful in it:
[Player.log](https://github.com/FluffyFishGames/AdvancedCompany/files/14093849/Player.log)
